.github/workflows/testing.yml

```diff
@@ -0,0 +1,30 @@
+name: pytest
+
+on:
+  pull_request
+
+jobs:
+  pytest:
+    runs-on: ubuntu-latest
+    steps:
+      # Check out repo and set up Python
+      - name: Check out the repository
+        uses: actions/checkout@v4
+        with:
+          lfs: true
+
+      # Use cached python and dependencies, install poetry
+      - name: "Setup Python, Poetry and Dependencies"
+        uses: packetcoders/action-setup-cache-python-poetry@main
+        with:
+          python-version: 3.8
+          poetry-version: 1.2.2
+
+      # Run pytest with coverage report and upload results to coveralls
+      - name: Run tests on python 3.8
+        run: |
+          poetry run pytest --cov=sed -n auto tests/
```

Comment:
This produces correct coverage for me locally: 
pytest --cov -n 10 -sv --show-capture=no tests/
Please add the other flags to track test output on the workflow, and remove the name from the cov flag to include tests into coverage calculation (at least that's our baseline)

Reply:
`poetry run pytest --cov=sed -n auto tests/`
gives
local:
<img width="472" alt="Screenshot 2023-10-17 at 16 17 26" src="https://github.com/OpenCOMPES/sed/assets/10053031/23070fe6-2e85-451b-8b10-e9320676e4eb">

on github actions:
<img width="566" alt="Screenshot 2023-10-17 at 16 20 17" src="https://github.com/OpenCOMPES/sed/assets/10053031/6297a522-dc6c-4154-b8b9-4348e5b2ee65">
also visible here: https://github.com/OpenCOMPES/sed/actions/runs/6533441246/job/17738634331


So I think that's also correct. But yes, adding the tests would be useful. --show-capture is useful to debug I'd say.
The problem seems not to be the coverage report but how the coveralls action is handling it. I am looking into that. It might be due to parallel pytest.

Reply:
Maybe just use the old way of uploading, and remove coverage from the multiversion test

Reply:
> --show-capture is useful to debug I'd say.

Yes, remove show capture, but maybe we should add --full-trace, this will not truncate trace backs on test failures, helpful to understand the cause